### DISCUSSION TITLE
Fitting Discontinuous Functions

### DISCUSSION THREAD

**Discussion by janzenchoi** [MilesCranmer/PySR#1006] [Q&A]
Hello,

I am using `TemplateExpressionSpec` to fit an expression (i.e., `z0`) that is undefined/complex after a certain input value (e.g., `x0 = L`). The template returns the relative error, which the loss function interprets as the loss (as per your previous suggestions). However, in fitting the expression (`z0`), the loss function may try to evaluate the relative error where the fitted expression is undefined (i.e., `x0 > L`).

Is there any way to configure the library's methods to have the loss function only conduct calculate the loss for defined / real values (i.e., `x0 < L`)? The following is a snippet of my code to fit the discontinuous expression.

```
expression_spec = TemplateExpressionSpec(
    expressions    = ["f0", "fA", "fn", "fM", "fphi", "fchi"],
    variable_names = ["x0", "x1", "x2", "y0"],
    combine        = """
        A = 10^fA(x2); n = abs(fn(x2)); M = 10^fM(x2); phi = abs(fphi(x2)); chi = abs(fchi(x2));
        z0 = A*x1^n * ((1-(phi+1)*M*x1^chi*x0)^((phi+1-n)/(phi+1))-1) / (M*x1^chi*(n-phi-1)) + f0(x0,x1,x2);
        ((y0-z0)/y0)^2
    """
)

regressor = PySRRegressor(
    expression_spec  = expression_spec,
    populations      = 32,
    population_size  = 32,
    maxsize          = 16,
    niterations      = 128,
    precision        = 64,
    verbosity        = 0,
    binary_operators = ["+", "*", "/"],
    unary_operators  = ["log"],
    elementwise_loss = "take_first(p, t, w) = p",
)
```

Thanks again for the help.
